### DISCUSSION TITLE
RFC: Theming

### DISCUSSION THREAD

**Discussion by pearofducks** [fabric-ds/issues#56] [Ideas]
https://github.com/fabric-ds/issues/pull/54

**Comment by benja** [comment]
Great write up. Instead of adding the core variables, would it be worth using something like [brightness](https://tailwindcss.com/docs/brightness)? E.g. for hover and active we'd just lower the brightness. It might just be easier to set up the core values though. Just thinking out loud.

> A background-primary will probably be different than a text-primary - I don't think this is that confusing to work with, probably just something we need to ensure we're clear on

How do we ensure these colors have enough contrast ratio? I assume most people will use something like black/coal-ish for text primary, and bg primary (provided by the consumer) may not always work with that in terms of contrast. Again, just thinking out loud.

| Background primary  | Text primary |
| ------------- | ------------- |
| Black  | White  |
| Black  | Black  |
| Blue  | White  |
| Blue  | Black  |
| Red | White |
| Red | Black |
...the list goes on

A lot of these depend on the level of blue, red, etc.

> Would we provide/inject the component-classes then instead of having them be imports? That way the components could take in this instanced set of classes?

Not sure how this would work in Vue, but in React we could export a context that all components get their classname values from. If the consumer doesn't set up the context provider with their custom classes, the component will default to what we already have setup.

The rest looks okay to me.

**Reply by pearofducks** [reply]
> Instead of adding the core variables, would it be worth using something like brightness?

We also need to support consumers doing the opposite - starting dark and hover/active being _lighter_. Otherwise this would be an OK way to go.

> How do we ensure these colors have enough contrast ratio? I assume most people will use something like black/coal-ish for text primary, and bg primary (provided by the consumer) may not always work with that in terms of contrast.

These would be up to the consumer to get right. But it's a great point that we should highlight in some eventual "theming docs" that they should ensure this happens! Also is something @adidick could highlight as these early conversations with consumers are happening.

> Not sure how this would work in Vue, but in React we could export a context...

Yeah Vue is basically the same, provide/inject I think is the analog to React's context!
